crates/image_viewer/src/image_info.rs

```diff
@@ -0,0 +1,147 @@
+use crate::ImageView;
+use anyhow;
+use gpui::{div, Context, Entity, IntoElement, ParentElement, Render, Subscription};
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use settings::{Settings, SettingsSources};
+use ui::{prelude::*, Button, LabelSize, Window};
+use workspace::{ItemHandle, StatusItemView, Workspace};
+
+#[derive(Clone, Debug, Serialize, Deserialize, JsonSchema, Default)]
+#[serde(rename_all = "snake_case")]
+pub enum ImageFileSizeUnitType {
+    #[default]
+    Binary,
+    Decimal,
+}
+
+impl Settings for ImageFileSizeUnitType {
+    const KEY: Option<&'static str> = Some("image_file_unit_type");
+
+    type FileContent = Self;
+
+    fn load(
+        sources: SettingsSources<Self::FileContent>,
+        _: &mut App,
+    ) -> Result<Self, anyhow::Error> {
+        sources.json_merge().or_else(|_| Ok(Self::Binary))
+    }
+}
+
+pub struct ImageInfo {
+    width: Option<u32>,
```

Comment:
As mentioned before, this should use the `ImageItemMeta` struct with an `Option`, rather than defining each field individually. Like:

```rs
pub struct ImageInfo {
  metadata: Option<ImageItemMeta>,
  _observe_active_image: Option<Subscription>
}
```

Reply:
updated.